torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -825,6 +825,8 @@ ProcessGroupNCCL::~ProcessGroupNCCL() {
 void ProcessGroupNCCL::ncclCommWatchdog() {
   try {
     LOG(INFO) << "[Rank " << rank_ << "] NCCL watchdog thread started!";
+    auto mode = cudaStreamCaptureModeThreadLocal;
```

Comment:
do we need to set back the thread to its previous mode after this is done ?

Reply:
I wouldn't think so unless the watchdog thread is further spawning additional threads or doing something other than simply polling enqueued work.